includes/class-wc-stripe-payment-information.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Class Payment_Information
+ *
+ * @package WooCommerce\Stripe
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Mostly a wrapper containing information on a single payment.
+ */
+class WC_Stripe_Payment_Information {
+
+	/**
+	 * Extract the payment token from the provided request.
+	 *
+	 * TODO: Once php requirement is bumped to >= 7.1.0 set return type to ?\WC_Payment_Token
+	 * since the return type is nullable, as per
+	 * https://www.php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration
+	 *
+	 * @param array $request Associative array containing payment request information.
+	 *
+	 * @return \WC_Payment_Token|NULL
+	 */
+	public static function get_token_from_request( array $request ) {
+		$payment_method    = $request['payment_method'] ? $request['payment_method'] : null;
```

Comment:
Is this `payment_method` field ever present during checkout? If my memory serves me faithfully, I believe this field was added in the old WCPay checkout flow, where before submitting the checkout form the client created a Stripe payment method and attached it to the checkout form as an additional field. This does not take place in the UPE checkout and instead this payment method is created by the payment element when it is confirmed. I think this line can be safely removed, but correct me if I'm mistaken.

Reply:
Good point, the `save payment method` is not part of this PR, the idea is to add it after adding the `add payment method` (in #1711), I'll remove this for now.